go-controller/pkg/ovn/egressgw.go

```diff
@@ -125,6 +241,7 @@ func (oc *Controller) addGWRoutesForNamespace(namespace string, egress gatewayIn
 		prefix, err := oc.extSwitchPrefix(pod.Spec.NodeName)
 		if err != nil {
 			klog.Infof("Failed to find ext switch prefix for %s %v", pod.Spec.NodeName, err)
+			routeUnlock()
 			continue
 		}
 
```

Comment:
@trozet I feel like the whole `for _, gw := range egress.gws {` block could benefit from being a separate function and using `defer routeUnlock()`

Reply:
good idea, will split it into it own function

Reply:
actually we already have addGWRoutesForPod, I can just call that